rita_common/src/rita_loop/write_to_disk.rs

```diff
@@ -0,0 +1,102 @@
+    save_client_settings, save_exit_settings,
+};
+
+pub const FAST_LOOP_TIMEOUT: Duration = Duration::from_secs(4);
+// Save duration for all writes to disk in order to reduce write operations
+// pub const SAVING_TO_DISK_FREQUENCY: Duration = Duration::from_secs(600);
+#[derive(Clone)]
+pub enum SettingsOnDisk {
+    RitaClientSettings(RitaClientSettings),
+    RitaExitSettingsStruct(RitaExitSettingsStruct),
+}
+/// This loop attempts to perform all write operations for writing to disk
+/// This includes writing config/settings, usage tracker, and debt tracker.
+/// It takes in a settings enum in order to identify what type the device
+/// is. There is also a consideration for the amount of storage the device
+/// has on disk since we don't want to save too often if the disk doesn't
+/// contain a lot of storage.
+pub fn save_to_disk_loop(mut old_settings: SettingsOnDisk, file_path: &str) {
+    let file_path = file_path.to_string();
+    let router_storage_small;
+    let saving_to_disk_frequency: Duration;
+
+    match old_settings.clone() {
+        SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+            match old_settings_client.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_client.save_interval);
+        }
+        SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+            match old_settings_exit.network.device.clone() {
+                Some(val) => router_storage_small = is_router_storage_small(&val),
+                None => router_storage_small = false,
+            };
+            saving_to_disk_frequency = Duration::from_secs(old_settings_exit.save_interval);
+        }
+    }
+
+    thread::spawn(move || loop {
+        let start = Instant::now();
+
+        if start.elapsed() < saving_to_disk_frequency {
+            thread::sleep(saving_to_disk_frequency - start.elapsed());
+        }
+
+        //settings
+        match old_settings.clone() {
+            SettingsOnDisk::RitaClientSettings(old_settings_client) => {
+                let new_settings = get_rita_client();
+                save_client_settings(old_settings_client, file_path.clone());
+                old_settings = SettingsOnDisk::RitaClientSettings(new_settings);
+            }
+            SettingsOnDisk::RitaExitSettingsStruct(old_settings_exit) => {
+                let new_settings = get_rita_exit();
+                save_exit_settings(old_settings_exit, file_path.clone());
+
+                old_settings = SettingsOnDisk::RitaExitSettingsStruct(new_settings);
+            }
+        }
+
+        // debt keeper
+        if !router_storage_small {
+            let mut dk = DEBT_DATA.write().unwrap();
+            trace!("sending debt keeper update");
+            dk.save_if_needed();
```

Comment:
to avoid making DEBT_DATA public just have save_if_needed as a independent function rather than a member function of the struct and it can grab the DEBT_DATA.write() there.